test/ContractTestContext.ts

```diff
@@ -90,107 +86,101 @@ export type ContractTestContext = ArchContracts & {
 export const signers = ethers.getSigners();
 export const ownerStartingLvUSDAmount = ethers.utils.parseUnits("1000.0");
 
-export async function buildContractTestContext (contractRoles: ContractRoles = {}): Promise<ContractTestContext> {
+/* reset the network and rebuild and deploy all contracts. accepts contractRoleOverrides which can specify
+   an overriden admin address for all contracts or individual roles per contract */
+export async function buildContractTestContext (contractRoleOverrides: ContractRoles = {}): Promise<ContractTestContext> {
     await helperResetNetwork(defaultBlockNumber);
 
     const context = {} as ContractTestContext;
 
     [context.owner, context.addr1, context.addr2, context.treasurySigner, context.addr3] = await signers;
 
-    const contractRolesWithDefaults = {
-        defaults: {
-            admin: context.owner.address,
-            executive: context.owner.address,
-            governor: context.owner.address,
-            guardian: context.owner.address,
-        },
-        ...contractRoles,
-    } as ContractRolesWithDefaults;
-
     context.externalOUSD = new ethers.Contract(addressOUSD, abiOUSDToken, context.owner);
     context.externalUSDT = new ethers.Contract(addressUSDT, abiUSDTToken, context.owner);
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     context.external3CRV = new ethers.Contract(address3CRV, abi3CRVToken, context.owner);
 
+    const signerAdmin = contractRoleOverrides.admin || context.owner;
     const contracts = await deployContracts<ArchContracts>({
-        archToken: ["ArchToken", context.owner.address],
-        cdp: ["CDPosition"],
-        coordinator: ["Coordinator"],
-        exchanger: ["Exchanger"],
-        leverageAllocator: ["LeverageAllocator"],
-        leverageEngine: ["LeverageEngine"],
-        lvUSD: ["LvUSDToken"],
-        parameterStore: ["ParameterStore"],
-        positionToken: ["PositionToken"],
-        vault: ["VaultOUSD", context.externalOUSD.address, "VaultOUSD", "VOUSD"],
-    }, contractRolesWithDefaults);
+        archToken: [context.owner.address],
+        cDPosition: [],
+        coordinator: [],
+        exchanger: [],
+        leverageAllocator: [],
+        leverageEngine: [],
+        lvUSDToken: [],
+        parameterStore: [],
+        positionToken: [],
+        vaultOUSD: [context.externalOUSD.address, "VaultOUSD", "VOUSD"],
+    }, signerAdmin.address);
```

Comment:
Idea here is to make all the instance variables the same as the contract path for simplicity and consistency.